include/Animal.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "Location.h"
+#include <iostream>
+#include <vector>
+#include <map>
+using namespace std;
+#include <cstdlib> // for rand()
+#include <ctime>   // for time()using namespace std;
+class Animal
+{
+protected:
+	string _name;
+	Location _location;
+	bool _stopped = true;
+
+public:
+	Animal(const string&, const Location&);
+	virtual void printDetails() const = 0;
+	virtual char getInitial() const = 0;
+	Location getLocation() const;
+	virtual void step() =0;
+	void stop();
```

Comment:
all functions should be pure not only the pure
read this
https://stackoverflow.com/questions/2391679/why-do-we-need-virtual-functions-in-c